layouts/default.vue

```diff
@@ -1,3 +1,34 @@
+<script setup lang="ts">
+type Theme = 'light' | 'dark'
+
+interface Message {
+  type: 'theme'
+  theme: Theme
+}
+
+function setTheme(theme: Theme) {
+  document.documentElement.classList.remove('light', 'dark')
+  document.documentElement.classList.add(theme)
+}
+
+const { initialTheme } = useRoute().query
+
+onMounted(() => {
+  if (initialTheme && (initialTheme === 'light' || initialTheme === 'dark')) {
```

Comment:
```suggestion
  if (initialTheme === 'light' || initialTheme === 'dark') {
```

Can't we not simplify it to this?

Another one is

```ts
['light','dark'].includes(initialTheme)
```

Reply:
Typescript can't type narrow with `includes` 😢, and the only reason why the first condition exists is because I tried to use `includes` in the first place 🤦 

Removed the first condition.